### PR TITLE
[SPIC-158] add: env meta data 

### PR DESCRIPTION
**Pull request by MicheilCathcart**

### CODE REVIEW

**Review by sebtoombs — COMMENTED** [review]
Just a question really (See comments).

Also; I notice you've put the inject env vars stuff in the "build" step?

a) does this even work? In theory (here's an example: https://dev.azure.com/mx51/spi-sample-pos/_build/results?buildId=101757&view=logs&s=e65ccba4-5b50-5543-f0ab-d89ddb8f0279) the "build" step should be run once for both deploy stages (i.e. you will get the same vars deployed to both deploy targets.

b) even if it did work, we might want to think about is it scalable? works for the "ENV" var (dev/live), but what if we want other vars beyond that in the future?

tl;dr - I think this needs to be in bot deploy dev and deploy live (I am unsure if there's an easy way to reuse the script or if just need to copy/paste)

File: azure-pipelines.yml [hunk 1]
```diff
@@ -61,6 +71,23 @@ stages:
               echo "$BUILD_SOURCEBRANCH" > build/version.txt
             displayName: 'Store git ref value in release package'
 
+          - script: |
+              echo "//registry.npmjs.org/:_authToken=$(NPM_TOKEN)" > ~/.npmrc
```
[the remark this answers is not in the document]
Comment (MicheilCathcart) [reply azure-pipelines.yml]: I believe this was writing the value, as the last iteration of this authed AOK. I also think it is available as ENV vars. Will double check it all, just assumed it worked AOK on last run

[the remark this answers is not in the document]
Comment (sebtoombs) [reply azure-pipelines.yml]: Ok, if it's writing the value of the token to the file we need to tweak - this is a security problem (could get leaked)
See the note under step 1 here: https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow#create-and-check-in-a-project-specific-npmrc-file

[the remark this answers is not in the document]
Comment (sebtoombs) [reply azure-pipelines.yml]: Just checked this, needs to be:

```
echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" > ~/.npmrc
```

**Review by sebtoombs — APPROVED** [review]
